### PR TITLE
Compositing operation implementation fixes

### PR DESCRIPTION
**Pull request by bernardkwok**
Fix for #400.
- Add in checks for possible divide by 0 for burn, dodge, disjoint over. Needed to add in explicit typed functions since this cannot be inlined. Also align so that it matches the Sony implementation.
- Add in edge case tests.
- Fix other tests cases so they have predictable results which are opaque to avoid differences when alpha blending kicks in for GLSL vs OSL which is premultiplied.

### CODE REVIEW

File: libraries/stdlib/genglsl/mx_disjointover_color2.glsl [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  if (summedAlpha <= 1)
+  {
+      result.x = fg.x + bg.x;
+  }
+  else
+  {
+      if (abs(bg.y) < M_FLOAT_EPS)
+      {
+        result.x = 0.0;
```
Comment (feldstj) [inline libraries/stdlib/genglsl/mx_disjointover_color2.glsl]: minor: missing a tab

File: libraries/stdlib/genglsl/mx_disjointover_color2.glsl [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      }
+      else
+      {
+        result.x = fg.x + ((bg.x * (1-fg.y)) / bg.y);
```
Comment (feldstj) [inline libraries/stdlib/genglsl/mx_disjointover_color2.glsl]: minor: missing a tab
